.github/workflows/galaxy-deepseek-tests-impl.yaml

```
@@ -67,23 +98,23 @@ jobs:
         with:
           build-artifact-name: ${{ inputs.build-artifact-name }}
           wheel-artifact-name: ${{ inputs.wheel-artifact-name }}
-      - name: Run DeepSeek tests
+      - name: Run DeepSeek unit tests
+        if: ${{ matrix.test-group.test-type == 'unit' }}
         timeout-minutes: ${{ matrix.test-group.timeout }}
         run: |
-          fail=0
-          echo "LOG_METAL: Running DeepSeek unit tests"
-          start_time=$(date +%s)
-
           pip install -r models/demos/deepseek_v3/reference/deepseek/requirements.txt
-          pytest models/demos/deepseek_v3/tests --timeout 10 --durations=0 || fail=1
-
-          end_time=$(date +%s)
-          duration=$((end_time - start_time))
-          echo "LOG_METAL: DeepSeek unit tests took $duration seconds to complete"
-
-          if (( fail )); then
-            exit 1
-          fi
+          pytest models/demos/deepseek_v3/tests/unit --timeout 60 --durations=0
+      - name: Validate weight cache
+        if: ${{ matrix.test-group.test-type == 'module' }}
+        timeout-minutes: 10
+        run: |
+          python3 models/demos/deepseek_v3/scripts/validate_weight_cache.py --root "$DEEPSEEK_V3_CACHE/test_cache" || true
```

```suggestion
          python3 models/demos/deepseek_v3/scripts/validate_weight_cache.py --root "$DEEPSEEK_V3_CACHE/test_cache"
```

The matrix will generate two separate jobs for this. Why are you doing `|| true`?

---

If you're good with that then sure...